src/text-area/text-area.tsx

```diff
@@ -139,5 +124,60 @@ const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(function T
     )
 })
 
+function useAutoExpand({
+    value,
+    autoExpand,
+    containerRef,
+    internalRef,
+}: {
+    value: string | undefined
+    autoExpand: boolean
+    containerRef: React.RefObject<HTMLDivElement>
+    internalRef: React.RefObject<HTMLTextAreaElement>
+}) {
+    const isControlled = value !== undefined
+
+    React.useEffect(
+        function setupAutoExpandWhenUncontrolled() {
+            const textAreaElement = internalRef.current
```

Comment:
💭 I see that in `setupAutoExpandWhenControlled` you check for `!isControlled -> return`.
Would it make sense to check for `isControl -> return` in this case?